core/domain/suggestion_services.py

```diff
@@ -542,3 +555,52 @@ def check_can_resubmit_suggestion(suggestion_id, user_id):
     suggestion = get_suggestion_by_id(suggestion_id)
 
     return suggestion.author_id == user_id
+
+
+def check_can_edit_suggestion(credentials_to_check, user_id):
```

Comment:
This function is slightly too complex. Basically, all you want to know is can the given user edit the suggestion. So ideally the output should be yes/no (boolean). 

Also, I think it might be a better idea to pass in the rights object from the above layer. That can reduce the complexity in this code.

Reply:
done